### DISCUSSION TITLE
Have two different DataSource for DatagridSelectColumn filtering and editing

### DISCUSSION THREAD

**Discussion by michele-bettin** [Megabit/Blazorise#5088] [Ideas]
Hi guys, I have seen in the documentation on that link https://blazorise.com/docs/extensions/datagrid/features/filtering that is possible to customize the filter dropdown of a DatagridSelectColumn to show a blank option to view all the genders in the datagrid.
Here is the code:
```
<DataGrid TItem="Employee"
          Data="@employeeList"
          Filterable
          Responsive>
    <DataGridColumn Field="@nameof( Employee.FirstName )" Caption="Name" Editable="false"></DataGridColumn>
    <DataGridSelectColumn CustomFilter="@OnGenderCustomFilter" TItem="Employee" Field="@nameof( Employee.Gender )" Caption="Gender" Editable Data="EmployeeData.Genders" ValueField="(x) => ((Gender)x).Code" TextField="(x) => ((Gender)x).Description" />
</DataGrid>

@code{
    private List<Employee> employeeList = new() { new() { FirstName = "David", Gender = "M" }, new() { FirstName = "Mladen", Gender = "M" }, new() { FirstName = "John", Gender = "M" }, new() { FirstName = "Ana", Gender = "F" }, new() { FirstName = "Jessica", Gender = "F" } };

    private bool OnGenderCustomFilter( object itemValue, object searchValue )
    {
        if ( searchValue is string genderFilter )
        {
            return genderFilter == "*" || genderFilter == itemValue?.ToString();
        }

        return true;
    }

}
```

Now if I add a "All Genders" option to view all the users regardless of their genders when I want to edit a row, in the edit form, I can also choose the"All Genders" for the gender, which is an error (since"All Genders" is not a gender).
So wouldn't be amazing if these two concepts will be separated and be able to exclude the"All Genders"from the editing...
Thanks...

**Comment by David-Moreira** [comment]
Hello,
I'ts just an example.
You should be able to override the `Filtertemplate`, present whatever component/values you like and then apply the custom filter over those values, since you get both an `itemValue` and a `filterValue` on the `CustomFilter` delegate.

**Comment by michele-bettin** [comment]
@David-Moreira Thanks! It worked!
